cyclonedx/model/bom.py

```diff
@@ -363,6 +364,38 @@ def has_vulnerabilities(self) -> bool:
         """
         return any(c.has_vulnerabilities() for c in self.components)
 
+    def validate(self) -> bool:
+        """
+        Perform data-model level validations to make sure we have some known data integrity prior to attempting output
+        of this `Bom`
+
+        Returns:
+             `bool`
+        """
+
+        # 1. Make sure dependencies are all in this Bom.
+        all_bom_refs = set([self.metadata.component.bom_ref] if self.metadata.component else []).union(
```

Comment:
did you know, that in python you could override operators?
implemented as 
so `set([1,2,3]) | set([2,3,4])` would create a new set with the union
so `set([1,2,3]) & set([2,3,4])` would create a new set with the difference/iintersection